src/windows/webview.c

```diff
@@ -547,7 +564,8 @@ struct webview *mty_webview_create(MTY_App *app, MTY_Window window, const char *
 	ctx->handler4.opaque = ctx;
 	ctx->opts.lpVtbl = &VTBL5;
 
-	const WCHAR *dirw = dir ? MTY_MultiToWideDL(dir) : L"webview-data";
+	WCHAR dirw[MTY_PATH_MAX] = {0};
```

Comment:
This would result in an array of 2000 WCHARs. Defining the array size as [size * sizeof(type)] is incorrect and exponential.

I would instead say that MTY_OpenFile has a bug that needs fixing.

Reply:
Strictly speaking, it wouldn't be exponential...it's just a linear function of the size of a char type.

The real question though is whether `MTY_PATH_MAX` is intended to be "max character count of a path" or "max bytes that any path can occupy". The comment in `matoya.h` says "Maximum size" but that's not helpful verbiage to clear the confusion.

If you can confirm/figure out the intent and then flag a potential bug in MTY_OpenFile (or fix your own code, if that's what you end up determining to be required), that would be great.

Reply:
If you double your type size, you quadruple your array size, but sure.

`MTY_PATH_MAX` is, to my understanding, a character count. so plain `MTY_PATH_MAX` is fine. If it were supposed to be a bytes size, I imagine it would be defined as `MTY_PATH_SIZE` or `MTY_PATH_BYTES`, or similar.

Reply:
> If you double your type size, you quadruple your array size, but sure.

Mea culpa. It would be linear if the declaration was `char file[MTY_PATH_MAX * sizeof(WCHAR)];`. Because the type of the variable is `WCHAR`, it is indeed exponential.

Which indeed points to a bug in `MTY_OpenFile`. Glad we hashed this out. We should fix it.